### PR TITLE
Swap Encoder `Builder` for Text over ByteString

### PR DESCRIPTION
**Pull request by mankyKitty**
There was a bug in how the characters in the `JString` were being converted to `Text` that would rear it's head when encountering literal character codes such as : `\128` or `\256`, among others.

The instances for `AsJString` for `Text` and `ByteString` have been removed as their `Prism`s were not correct. The `Text` has been replaced with a more correct `Prism` and the `ByteString.Char8` is now handled with a couple of functions instead.

A regression test has been added to try to catch this behaviour if it appears again. Additionally more property tests have been added to ensure we can round trip unicode, latin1, and ascii for both `Text` and `ByteString.Char8` types.

### CODE REVIEW

File: src/Waargonaut/Types/JChar.hs [hunk 1]
```diff
@@ -127,6 +132,7 @@ instance AsJCharUnescaped Char where
     where
       excluded =
         [ (== '\NUL')
+        , (== '\DEL')
```
Comment (benkolera) [inline src/Waargonaut/Types/JChar.hs]: Was this actually still necessary?

Comment (mankyKitty) [reply src/Waargonaut/Types/JChar.hs]: Nope! :tada:
